plugins/openwhisk/src/main/java/com/navercorp/pinpoint/plugin/openwhisk/OpenwhiskPlugin.java

```diff
@@ -0,0 +1,225 @@
+                final InstrumentClass target = instrumentor.getInstrumentClass(classLoader, className, classfileBuffer);
+
+                // add `started` method interceptor
+                final MethodFilter startedMethodFilter = MethodFilters.chain(
+                        MethodFilters.name("started$extension")
+                );
+                for (InstrumentMethod method : target.getDeclaredMethods(startedMethodFilter)) {
+                    try {
+                        method.addInterceptor("com.navercorp.pinpoint.plugin.openwhisk.interceptor.TransactionIdStartedInterceptor");
+                    } catch (Exception e) {
+                        if (logger.isWarnEnabled()) {
+                            logger.warn("Unsupported method " + method, e);
+                        }
+                    }
+                }
+
+                // add `finished` method interceptor
+                final MethodFilter finishedMethodFilter = MethodFilters.chain(
+                        MethodFilters.name("finished$extension")
+                );
+                for (InstrumentMethod method : target.getDeclaredMethods(finishedMethodFilter)) {
+                    try {
+                        method.addInterceptor("com.navercorp.pinpoint.plugin.openwhisk.interceptor.TransactionIdFinishedInterceptor");
+                    } catch (Exception e) {
+                        if (logger.isWarnEnabled()) {
+                            logger.warn("Unsupported method " + method, e);
+                        }
+                    }
+                }
+
+                // add `failed` method interceptor
+                final MethodFilter failedMethodFilter = MethodFilters.chain(
+                        MethodFilters.name("failed$extension"),
+                        MethodFilters.argAt(2, "whisk.common.StartMarker"),
+                        MethodFilters.argAt(3, "scala.Function0")
+                );
+                for (InstrumentMethod method : target.getDeclaredMethods(failedMethodFilter)) {
+                    try {
+                        method.addInterceptor("com.navercorp.pinpoint.plugin.openwhisk.interceptor.TransactionIdFailedInterceptor");
+                    } catch (Exception e) {
+                        if (logger.isWarnEnabled()) {
+                            logger.warn("Unsupported method " + method, e);
+                        }
+                    }
+                }
+
+                // add `mark` method interceptor
+                final MethodFilter markMethodFilter = MethodFilters.chain(
+                        MethodFilters.name("mark$extension")
+                );
+                for (InstrumentMethod method : target.getDeclaredMethods(markMethodFilter)) {
+                    try {
+                        method.addInterceptor("com.navercorp.pinpoint.plugin.openwhisk.interceptor.TransactionIdMarkInterceptor");
```

Comment:
If you are checking the number of method parameters after adding an Interceptor, you do not need to check the arguments.
If not, check the number of arguments. (to prepare for overloading .)